pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,521 @@
+			{Model: endpoint.APIParamModelText, Name: "name", Required: true},
+			{Model: endpoint.APIParamModelText, Name: "value", Required: true},
+		},
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_members_show",
+		Component: model.NodeKindPD,
+		Path:      "/members",
+		Method:    endpoint.MethodGet,
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_leader_show",
+		Component: model.NodeKindPD,
+		Path:      "/leader",
+		Method:    endpoint.MethodGet,
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_operator_show",
+		Component: model.NodeKindPD,
+		Path:      "/operators",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "kind"},
+		},
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_regions",
+		Component: model.NodeKindPD,
+		Path:      "/regions",
+		Method:    endpoint.MethodGet,
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_region_id",
+		Component: model.NodeKindPD,
+		Path:      "/region/id/{regionID}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "regionID", Required: true},
+		},
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_region_key",
+		Component: model.NodeKindPD,
+		Path:      "/region/key/{regionKey}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "regionKey", Required: true},
+		},
+	}, endpoint.MiddlewareHandlerFunc(func(req *endpoint.Request) error {
+		req.PathValues.Set("regionKey", url.QueryEscape(req.PathValues.Get("regionKey")))
+		return nil
+	}))
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_region_scan",
+		Component: model.NodeKindPD,
+		Path:      "/regions/key",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "key", Required: true},
+			{Model: endpoint.APIParamModelInt, Name: "limit", Required: true},
+		},
+	}, endpoint.MiddlewareHandlerFunc(func(req *endpoint.Request) error {
+		req.PathValues.Set("key", url.QueryEscape(req.PathValues.Get("key")))
+		return nil
+	}))
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_region_sibling",
+		Component: model.NodeKindPD,
+		Path:      "/regions/sibling/{regionID}",
+		Method:    endpoint.MethodGet,
+		PathParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "regionID", Required: true},
+		},
+	})
+	c.AddEndpoint(&endpoint.APIModel{
+		ID:        "pd_region_start_key",
+		Component: model.NodeKindPD,
+		Path:      "/regions/key",
+		Method:    endpoint.MethodGet,
+		QueryParams: []*endpoint.APIParam{
+			{Model: endpoint.APIParamModelText, Name: "key", Required: true},
+			{Model: endpoint.APIParamModelInt, Name: "limit"},
+		},
+	}, endpoint.MiddlewareHandlerFunc(func(req *endpoint.Request) error {
+		req.PathValues.Set("key", url.QueryEscape(req.PathValues.Get("key")))
```

Comment:
How about making `endpoint.MiddlewareHandlerFunc` to be `OnReceiveRequest`? Then `AddEndpoint` may be not necessary any more!